stl/inc/format

```diff
@@ -2851,6 +2946,8 @@ template <class _Ty, class _CharT, _Basic_format_arg_type _ArgType>
 struct _Formatter_base {
     using _Pc = basic_format_parse_context<_CharT>;
 
+    using _Is_formatter_base_specialization = void;

```

Comment:
I think it's useful to account for width and precision from user defined `formatter`s that derive from standard `formatter`s. See https://github.com/microsoft/STL/pull/2437#discussion_r794359911. Also can't use `is_base_of` because `_Formatter_base` is a template.

Reply:
yeah, the way to go is to do the `_From_primary` thing that `allocator` does.

I guess I agree that it's useful, but it adds a footgun if we keep the exact estimation paths

Reply:
> I guess I agree that it's useful, but it adds a footgun if we keep the exact estimation paths

I don't think users can specialize `formatter` for `basic_string_view`, `basic_string` or `(const) char*` so there isn't really any footgun.